.changeset/rare-hornets-rush.md

```diff
@@ -0,0 +1,5 @@
+---
+"@fuel-ts/utils": patch
+---
+
+chore: integrate vitest matchers globally
```

Comment:
@nedsalk According to previous conversations, please add two things to the PR description:
1. Flowchart describing the general flow ([ref](https://github.com/FuelLabs/fuels-ts/pull/3462))
2. Task list of everything that is pending, like:
   - [ ] Task 1
   - [ ] Task 2
   - [ ] Etc

Reply:
Please also add some comprehensive examples using the API.

The ones we have there seem too brief and don't give the complete picture of what's possible.

Reply:
I expected a flowchart with a start and end describing the implementation flow to facilitate the reviews, possibly using Mermaid for easy collaboration. It's fragmented and seems backward in that one must read the code to understand the flowchart and not the other way around.

We chose to have a parsing layer to help consumers avoid all the complexities of the ABI spec, and the API is the heart of what matters to them. However, the test you mentioned and the docs you wrote seem very basic and don't explain anything. I'll have to dig deeper and play with the code.